proposals/array_set.json

```diff
@@ -0,0 +1,215 @@
+{
+    "id": "array_set",
+    "summary": "Change the content of an array (insert, remove, update)",
+    "description": "Allows to insert into, remove from or update an array.\n\nAll labels get discarded and the array indices are always a sequence of numbers with the step size of 1 and starting at 0.",
+    "categories": [
+        "arrays"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "An array.",
+            "schema": {
+                "type": "array",
+                "items": {
+                    "description": "Any data type is allowed."
+                }
+            }
+        },
+        {
+            "name": "values",
+            "description": "The values to fill the array with.",
+            "schema": {
+                "type": "array",
+                "items": {
+                    "description": "Any data type is allowed."
+                }
+            }
+        },
+        {
+            "name": "index",
+            "description": "The index of the element to insert the value(s) before. To insert after the last element, specify the number of elements in the array. If the index is greater than the number of elements, the array is filled with `null` (no-data) values until the index is reached and then the values are added starting at the index given. The number of elements can be retrieved with the process ``count()`` having the parameter `condition` set to true.",
```

Comment:
> If the index is greater than the number of elements, the array is filled with `null` (no-data) values until the index is reached and then the values are added starting at the index given.

This sounds like  a non-neglectable feature of this function and has functional overlap with array_create. It doesn't feel very clean to me. I would throw an error on index overflow.

Reply:
It's more array_merge that can also solve this use case, right? Indeed, we could simplify this process and instead let users use array_merge (+ array_create) instead.

Reply:
I'm happy to go either route (leave it in or remove it). Maybe it would be good to have another opinion? @aljacob @lforesta ...

Reply:
I've simplified array_modify as requested. @soxofaan